python_modules/dagster/dagster/_core/definitions/asset_spec.py

```diff
@@ -76,18 +75,30 @@ def __new__(
         freshness_policy: Optional[FreshnessPolicy] = None,
         auto_materialize_policy: Optional[AutoMaterializePolicy] = None,
     ):
-        dep_set = set()
+        from dagster._core.definitions.asset_dep import AssetDep
+
+        dep_set = {}
         if deps:
             for dep in deps:
-                if isinstance(dep, AssetSpec):
-                    dep_set.add(dep.asset_key)
+                if not isinstance(dep, AssetDep):
+                    asset_dep = AssetDep(dep)
                 else:
-                    dep_set.add(AssetKey.from_coercible_or_definition(dep))
+                    asset_dep = dep
```

Comment:
"from_coercible" on AssetDep would be good

Reply:
also going to punt on this since i think the future of `CoercibleToAssetKey` will affect the naming of this method